core/include/scipp/core/element/histogram.h

```diff
@@ -32,15 +32,25 @@ namespace histogram_detail {
 template <class Out, class Coord, class Weight, class Edge>
 using args = std::tuple<span<Out>, span<const Coord>, span<const Weight>,
                         span<const Edge>>;
+
+template <typename T> constexpr auto get_value(const T &x) noexcept {
+  if constexpr (std::is_same_v<std::remove_reference_t<T>, time_point>) {
+    return x.time_since_epoch();
+  } else {
+    return x;
+  }
 }
+} // namespace histogram_detail
 
 static constexpr auto histogram = overloaded{
-    element::arg_list<histogram_detail::args<float, double, float, double>,
-                      histogram_detail::args<float, int64_t, float, double>,
-                      histogram_detail::args<double, double, double, double>,
-                      histogram_detail::args<double, float, double, double>,
-                      histogram_detail::args<double, float, double, float>,
-                      histogram_detail::args<double, double, float, double>>,
+    element::arg_list<
+        histogram_detail::args<float, double, float, double>,
+        histogram_detail::args<float, int64_t, float, double>,
+        histogram_detail::args<double, double, double, double>,
+        histogram_detail::args<double, float, double, double>,
+        histogram_detail::args<double, float, double, float>,
+        histogram_detail::args<double, double, float, double>,
+        histogram_detail::args<double, time_point, double, double>>,
```

Comment:
Wouldn't we want to bin into `time_point` edges rather than `double`? Is this just for plotting? I think we cannot really justify supporting this here, since it will also allow manual binning of datetimes into double edges, which is not really meaningful conceptually (unless we think of the underlying time since epoch).

Reply:
True, fixed it. This even made the histogram code simpler.